### PR TITLE
mousai: init at 0.3.1

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change

Package music recognition tool [mousai](https://github.com/SeaDve/Mousai)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/mousai/default.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+, gtk3
+, desktop-file-utils
+, libhandy
+, gst_all_1
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "mousai";
+  version = "0.2.0";
+  strictDeps = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:18]:
```suggestion
```

this is the default right now.

File: pkgs/applications/audio/mousai/default.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/SeaDve/Mousai/archive/refs/tags/v${version}.tar.gz";
+    sha256 = "1lzlrj1q2vcg6k9gilxj2b3z6d0iaqwn60iwi48v5blwjdaqr96v";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:29]:
```suggestion
    pkg-config
```

File: pkgs/applications/audio/mousai/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
+{ fetchurl
+, stdenv
+, meson
+, ninja
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:5]:
```suggestion
, pkg-config
```

File: pkgs/applications/audio/mousai/default.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  meta = with lib; {
+    description = "Simple application for identifying songs";
+    homepage = "https://github.com/SeaDve/Mousai";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ onny ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:66]: One new line to much.

File: pkgs/applications/audio/mousai/default.nix [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    gobject-introspection
+    libhandy
+  ] ++ (with gst_all_1; [
+    gstreamer
+    gst-plugins-good
+    gst-plugins-ugly
+    gst-plugins-base
+  ]);
+
+  propagatedBuildInputs = with python3.pkgs; [
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:47]: Wrapping the binaries is preferred.

File: pkgs/applications/audio/mousai/default.nix [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+, gst_all_1
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "mousai";
+  version = "0.2.0";
+  strictDeps = false;
+  format = "other";
+
+  src = fetchurl {
```
[the remark this answers is not in the document]
Comment (onny) [reply pkgs/applications/audio/mousai/default.nix:21]: I tried this before but somehow fetchFromGitHub tries to download file at
```
https://github.com/SeaDve/Mousai/archive/0.2.0.tar.gz
```
Where as the file is located at
```
https://github.com/SeaDve/Mousai/archive/refs/tags/v0.2.0.tar.gz
```

File: pkgs/applications/audio/mousai/default.nix (hunk 5 above)
Comment (onny) [reply pkgs/applications/audio/mousai/default.nix:47]: What is the best way to do this? :)

File: pkgs/applications/audio/mousai/default.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (fabaff) [reply pkgs/applications/audio/mousai/default.nix:21]: rev = "v${version}";` should help.

**Comment by onny** [comment]
Thank you @fabaff and @SuperSandro2000 for the feedback. All remarks should be fixed now :)

File: pkgs/applications/audio/mousai/default.nix [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  ];
+
+  preFixup = ''
+    wrapPythonPrograms
+  '';
+
+  postPatch = ''
+    chmod +x build-aux/meson/postinstall.py
+    patchShebangs build-aux/meson/postinstall.py
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:62]:
```suggestion
```

File: pkgs/applications/audio/mousai/default.nix [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  pname = "mousai";
+  version = "0.2.0";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "SeaDve";
+    repo = "Mousai";
+    rev = "v${version}";
+    sha256 = "10g3wj9yy15yzyfa1l96i97i3k6blm064p49hn7hp001cmc3w045";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:25]:
```suggestion
  };

  postPatch = ''
    chmod +x build-aux/meson/postinstall.py
    patchShebangs build-aux/meson/postinstall.py
  '';
```

File: pkgs/applications/audio/mousai/default.nix [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pygobject3
+    gst-python
+    requests
+  ];
+
+  preFixup = ''
+    wrapPythonPrograms
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mousai/default.nix:62]: We need to do something about this tripple wrapping.

```ShellSession
$ results/mousai/bin/
mousai                             .mousai-wrapped_                   ...mousai-wrapped-wrapped-wrapped
.mousai-wrapped                    ..mousai-wrapped-wrapped
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/mousai/default.nix (hunk 9 above)
Comment (onny) [reply pkgs/applications/audio/mousai/default.nix:70]: I managed to have it only double-wrapped
```
ls /nix/store/pgf869n0wjb2k9s31p1q2858wwkncijl-mousai-0.2.0/bin -a
.  ..  mousai  .mousai-wrapped  ..mousai-wrapped-wrapped
```
Took the concept from here https://github.com/NixOS/nixpkgs/blob/a230fd4368bf8d3db02800b33cbb550d21a5a1aa/pkgs/applications/misc/openlp/default.nix

**Comment by onny** [comment]
Not sure about the wrapping issue but maybe it is okay at this stage :)

**Comment by SuperSandro2000** [comment]
> Not sure about the wrapping issue but maybe it is okay at this stage :)

Looks better but still double wrapped

```
.r-xr-xr-x root root 2.8 KB Thu Jan  1 01:00:01 1970  ..mousai-wrapped-wrapped
.r-xr-xr-x root root 4.4 KB Thu Jan  1 01:00:01 1970  .mousai-wrapped
.r-xr-xr-x root root 3.6 KB Thu Jan  1 01:00:01 1970  mousai
```

**Comment by onny** [comment]
Sorry, I moved my local changes to a separate branch. New pull request is here: https://github.com/NixOS/nixpkgs/pull/120515
Hopefully it should also solve the issue of nested wrappers :)

**Pull request closed without merging**